File: DTOs/Sources/PosterServiceDTOs/UpdatePosterPositionDTO.swift
import Foundation

public struct UpdatePosterPositionDTO: Codable {
    public var posterId: UUID?
    public var latitude: Double?
    public var longitude: Double?
    public var expiresAt: Date?
    public var responsibleUsers: [UUID]?
    public var image: Data?
    
    public init(posterId: UUID? = nil, latitude: Double? = nil, longitude: Double? = nil, imageUrl: String? = nil, expiresAt: Date? = nil, responsibleUsers: [UUID]? = nil, image: Data? = nil) {
        self.latitude = latitude 
        self.longitude = longitude 
        self.posterId = posterId
        self.expiresAt = expiresAt
        self.responsibleUsers = responsibleUsers
        self.image = image
    }
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/PosterServiceDTOs/UpdatePosterPositionDTO.kt
package net.ipv64.kivop.dtos.PosterServiceDTOs

import java.util.UUID
import java.time.LocalDateTime

data class UpdatePosterPositionDTO (
    var posterId : UUID?,
    var latitude : Double?,
    var longitude : Double?,
    var expiresAt : LocalDateTime?,
    var responsibleUsers : List<UUID>?,
    var image : ByteArray?,
)
